### PR TITLE
Lifecycle session async

### PR DESCRIPTION
**Pull request by kpears201**

### CODE REVIEW

File: src/openrpc/lifecycle_management.json [hunk 1]
```diff
@@ -85,7 +85,25 @@
...
-                    "$ref": "#/components/schemas/AppSession"
+                    "oneOf": [
+                        {
+                            "$ref": "#/components/schemas/AppSession"
+                        },
+                        {
+                            "type": "object",
+                            "properties": {
+                                "state": {
+                                    "const": "pending"
```
Comment (Vinodsathyaseelan) [inline src/openrpc/lifecycle_management.json:96]: Is it required to define "state" in the schema with all possible const values (pending/created/anyother_state) ?

File: src/openrpc/lifecycle_management.json [hunk 2]
```diff
@@ -64,7 +64,7 @@
...
+            "summary": "Called before an app is launched in order to set up the lifecycle session. The session should be given an intent that will be passed to the application. This will behave differently depending on the current state of the application. If the application is already loaded but not active then a new active session is started. The intent is given to the application via the onNavigateTo event. If the app is not loaded, then it is considered a 'cold start'. A new loaded session will be created and the intent will be stored for the app to access via Parameters.initialization. Before a session is created for the application some checks may take place. For instance, if a user grant is required for the application then the user grant will be resolved before the session is created. In this case, this API will return immediately with a state='pending'. The created session will then be sent as an event in the LifecycleManagement.onSessionCreated event. If the pre-session checks fail, such as a user deferring a grant, then the LifecycleManagement.onSessionCanceled event will be emitted.",
```
Comment (Vinodsathyaseelan) [inline src/openrpc/lifecycle_management.json:67]: What is the workflow for the app to register for these events ? I mean when do the app register for these events ?

Comment (Vinodsathyaseelan) [reply src/openrpc/lifecycle_management.json:67]: Can you also confirm  that the GrantPolicy changes with evaluateAt is updated in Firebolt-open-rpc ?

I found a scenario where in 'cold start' mode with launch state active, both loaded_session_id & active_session_id will be created anew.  

Is there any GrantPolicy requirement to evaluate user grant at a new ActiveSession OR LoadedSession ?

Comment (kpears201) [reply src/openrpc/lifecycle_management.json:67]: Apps should register for these events when they start up

Comment (kpears201) [reply src/openrpc/lifecycle_management.json:67]: I will add evaluateAt to the GrantPolicy spec

File: src/openrpc/lifecycle_management.json (hunk 1 above)
Comment (kpears201) [reply src/openrpc/lifecycle_management.json:96]: There are two possible values, pending and created. Those are the two consts in the oneOf.
If state is pending, that means sessionId, loadedSessionId and activeSessionId must all be undefined.
If state is created then it follows the spec of AppSession, which says that appId, sessionId and loadedSessionId must be defined.

**Review by Vinodsathyaseelan — APPROVED** [review]

**Pull request merged**